addons/account/models/account_bank_statement.py

```diff
@@ -915,7 +923,12 @@ def process_reconciliation(self, counterpart_aml_dicts=None, payment_aml_rec=Non
             if abs(total)>0.00001:
                 partner_id = self.partner_id and self.partner_id.id or False
                 partner_type = False
+                payment_type = False
                 if partner_id:
+                    if len(account_types) == 1:
+                        partner_type = 'customer' if account_types == receivable_account_type else 'supplier'
+                        payment_type = 'inbound' if account_types == receivable_account_type else 'outbound'
```

Comment:
The payment type does not relate to the payable/receivable account but the positive or negative money flow, so I think this don't need to be touched with the fix.